extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1225,50 +1223,78 @@ static public function getLocalEnvURL( $url, $forcedEnv = null ) {
-		if ( !empty( $wgDevDomain ) ) {
-			$server = str_replace( ".{$wgDevDomain}", '', $server );
+
+		$devboxRegex = '/\.([^\.]+)\.wikia-dev\.(us|pl|com)$/';
+		if ( preg_match( $devboxRegex, $server, $groups ) === 1 ) {
+			$devBox = $groups[1];
+		} else {
+			$devBox = '';
 		}
+
+		$server = str_replace( '.' . $devBox . '.wikia-dev.com', '', $server );
 		$server = str_replace( static::WIKIA_TOP_DOMAIN, '', $server );
-		$server = str_replace( '.' . $wgWikiaBaseDomain, '', $server ); // PLATFORM-2400: make WF redirects work on staging
-
-		// determine the environment we want to get url for
-		$environment = (
-			$forcedEnv &&
-			$forcedEnv !== WIKIA_ENV_DEV &&
-			$forcedEnv !== WIKIA_ENV_SANDBOX
-		) ? $forcedEnv : $wgWikiaEnvironment;
-
-		// put the address back into shape and return
-		// we need to change 'https' to 'http' for stable/preview/verify/sandbox environments cause
-		// we do not have valid ssl certificate for these subdomains
-		switch ( $environment ) {
-			case WIKIA_ENV_PREVIEW:
-				return 'http://preview.' . $server . static::WIKIA_TOP_DOMAIN . $address;
-			case WIKIA_ENV_VERIFY:
-				return 'http://verify.' . $server . static::WIKIA_TOP_DOMAIN . $address;
-			case WIKIA_ENV_STABLE:
-				return 'http://stable.' . $server . static::WIKIA_TOP_DOMAIN . $address;
-			case WIKIA_ENV_STAGING:
-			case WIKIA_ENV_PROD:
-				return sprintf( '%s://%s.%s%s', $protocol, $server, $wgWikiaBaseDomain, $address );
-			case WIKIA_ENV_SANDBOX:
-				return 'http://' . static::getExternalHostName() . '.' . $server .
-				       static::WIKIA_TOP_DOMAIN . $address;
-			case WIKIA_ENV_DEV:
-				return 'http://' . $server . '.' . $wgDevDomain . $address;
-		}
-
-		throw new Exception( sprintf( '%s: %s', __METHOD__, 'unknown env detected' ) );
+
+		// PLATFORM-2400: make WF redirects work on staging
+		$server = str_replace( '.' . F::app()->wg->WikiaBaseDomain, '', $server );
+
+		// Determine the environment we want to get url for
+		$environment = self::getCurrentEnvironment( $forcedEnv );
+
+		// put the path back into shape and return
+		if ( $environment == WIKIA_ENV_PREVIEW ) {
+			$fullName = 'preview.' . $server . static::WIKIA_TOP_DOMAIN;
+		} elseif ( $environment == WIKIA_ENV_VERIFY ) {
+			$fullName = 'verify.' . $server . static::WIKIA_TOP_DOMAIN;
+		} elseif ( $environment == WIKIA_ENV_STABLE ) {
+			$fullName = 'stable.' . $server . static::WIKIA_TOP_DOMAIN;
+		} elseif ( $environment == WIKIA_ENV_PROD || $environment == WIKIA_ENV_STAGING ) {
+			$fullName = $server . '.' . F::app()->wg->WikiaBaseDomain;
+		} elseif ( $environment == WIKIA_ENV_SANDBOX ) {
+			$fullName = static::getExternalHostName() . '.' . $server . static::WIKIA_TOP_DOMAIN;
+		} elseif ( $environment == WIKIA_ENV_DEV) {
+			$fullName = $server . '.' . F::app()->wg->DevDomain;
+		} else {
+			throw new Exception( sprintf( '%s: %s', __METHOD__, 'unknown env detected' ) );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)